pkg/process/metadata/parser/service.go

```diff
@@ -0,0 +1,207 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package parser
+
+import (
+	"path/filepath"
+	"strings"
+	"unicode"
+
+	ddconfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/process/metadata"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type serviceExtractorFn func(args []string) string
+
+// List of binaries that usually have additional process context of whats running
+var binsWithContext = map[string]serviceExtractorFn{
+	"python":    parseCommandContextPython,
+	"python2.7": parseCommandContextPython,
+	"python3":   parseCommandContextPython,
+	"python3.7": parseCommandContextPython,
+	"ruby2.3":   parseCommandContext,
+	"ruby":      parseCommandContext,
+	"java":      parseCommandContext,
+	"sudo":      parseCommandContext,
+}
+
+var _ metadata.Extractor = &ServiceExtractor{}
+
+// ServiceExtractor infers a service tag by extracting it from a process
+type ServiceExtractor struct {
+	Enabled      bool
+	serviceByPID map[int32]*serviceMetadata
```

Comment:
Should we think about cleaning it - this is unbounded right now.

Reply:
Maybe can use [simplelru](https://github.com/hashicorp/golang-lru/blob/v2.0.0/simplelru/lru.go) here - CWS uses it .
Most common LRU usages in the Agent are either `simplelru` (before generics) or https://github.com/golang/groupcache/blob/master/lru/lru.go.

Reply:
I've refactored this to recreate a map on each call to extract, so this is no longer an issue.